contracts/StablecoinConverter.sol

```diff
@@ -175,18 +186,35 @@ contract StablecoinConverter is EpochTokenLocker {
                 volumes[i]
             );
         }
+        int fee = tokenConservation[0];
+        require(fee < auctionData[batchIndex].feeCollected, "Fee is not higher than before");
+        auctionData[batchIndex].feeCollected = fee;
+        auctionData[batchIndex].bestSolutionSubmitter = msg.sender;
         checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
 
     function checkTokenConservation(
         int[] memory tokenConservation
     ) internal pure {
-        for (uint i = 0; i < tokenConservation.length; i++) {
+        for (uint i = 1; i < tokenConservation.length; i++) {
             require(tokenConservation[i] == 0, "Token conservation does not hold");
         }
     }
 
+    function updateCurrentPrices(
+        uint128[] memory prices,  //list of prices for touched token only, frist price is fee Token price
+        uint16[] memory tokenIdsForPrice  // price[i] is the price for the token with tokenID tokenIdsForPrice[i]
+    ) internal {
+        for (uint i = 0; i < previousSolution.tokenIdsForPrice.length; i++) {
+            currentPrices[previousSolution.tokenIdsForPrice[i]] = 0;
+        }
+        currentPrices[0] = prices[0];
+        for (uint i = 0; i < tokenIdsForPrice.length; i++) {
```

Comment:
Does it make sense to start from 0 here?

Reply:
Yes, as we also want to cover: tokenIdsForPrice[0], which is not the fee token